frontend/src/components/utils/AssignStoryModal.tsx

```diff
@@ -0,0 +1,262 @@
+import { ApprovedLanguagesMap } from "../../utils/Utils";
+
+export type StoryToAssign = {
+  storyId: number;
+  storyTranslationId?: number;
+  title: string;
+  description: string;
+  level: number;
+  language: string;
+};
+
+export type AssignStoryModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onAssignStory: (story: StoryToAssign) => void;
+  approvedLanguagesTranslation: ApprovedLanguagesMap;
+  approvedLanguagesReview: ApprovedLanguagesMap;
+};
+
+const DropdownIndicator = () => {
+  return (
+    <Box
+      borderLeft="5px solid transparent"
+      borderRight="5px solid transparent"
+      borderTop="9px solid black"
+      margin="10px"
+    />
+  );
+};
+
+const colourStyles: StylesConfig = {
+  control: (styles) => ({
+    ...styles,
+    backgroundColor: "#F1F4F7", // #F1F4F7 is gray.100
+    border: "none",
+  }),
+  option: (styles, { isSelected }) => {
+    return {
+      ...styles,
+      color: "black",
+      backgroundColor: isSelected ? "#F1F4F7" : "white", // #F1F4F7 is gray.100
+      "&:hover": {
+        backgroundColor: "#F1F4F7", // #F1F4F7 is gray.100
+      },
+    };
+  },
+  indicatorSeparator: (styles) => ({ ...styles, display: "none" }),
+  placeholder: (styles) => ({ ...styles, color: "black" }),
+};
+
+const AssignStoryModal = ({
+  isOpen,
+  onClose,
+  onAssignStory,
+  approvedLanguagesTranslation,
+  approvedLanguagesReview,
+}: AssignStoryModalProps) => {
+  const [role, setRole] = useState<string | null>(null);
+  const [language, setLanguage] = useState<string | null>(null);
+  const [level, setLevel] = useState<number | null>(null);
+  const [stories, setStories] = useState<StoryToAssign[] | null>(null);
+  const [selectedStory, setSelectedStory] = useState<StoryToAssign | null>(
+    null,
+  );
+
+  const query = buildAssignStoryQuery(role === "Translator", language, level);
+
+  useQuery(query.string, {
+    fetchPolicy: "cache-and-network",
+    skip: language == null || level == null,
+    onCompleted: (data) => {
+      const storyData = data[query!!.fieldName].map((storyObj: any) => ({
+        ...storyObj,
+        language,
+      }));
+      setStories(storyData);
+    },
+  });
+
+  useEffect(() => {
+    // reset story selection when one of the fields change
+    setSelectedStory(null);
+  }, [language, level, role]);
+
+  const approvedLanguages =
+    role === "Translator"
+      ? approvedLanguagesTranslation
+      : approvedLanguagesReview;
+
+  const languageOptions = role
+    ? Object.keys(approvedLanguages).map((value) => ({
+        value,
+      }))
+    : [];
+
+  // display levels that the user is approved for
+  const levelOptions = language
+    ? [...Array(approvedLanguages[language] + 1).keys()]
+        .slice(1)
+        .map((value) => ({ value: `${value}` }))
+    : [];
+
+  const disabledStyle = useStyleConfig("Disabled");
+  const isPropertySelectDisabled = role == null;
+  const isTitleSelectDisabled = language == null || level == null;
```

Comment:
nit: theres a gap of time between the language&level being non null and having the actual titles retrieved from our api. in this time gap, the title select is enabled despite a lack of titles. Can we make this check on the length of the `stories` in addition to language&level being selected?

Reply:
@gaoxk if we add the check for length then the title select will be disabled if there are no stories to show, would that be misleading to the user (as opposed to having a dropdown with no options)?

Reply:
hmmm I see what you mean. I think this would be effective and consider your described case:
`const isTitleSelectDisabled = (language == null || level == null) && stories.length > 0;`